### PR TITLE
Bonsai & assets

### PR DESCRIPTION
**Pull request by apaskulin**
## Description

- [x] Updates the asset guide to use Bonsai
- [x] Better description of assets in asset reference

### CODE REVIEW

File: content/sensu-go/5.2/guides/install-check-executables-with-assets.md [hunk 1]
```diff
@@ -10,153 +12,119 @@ menu:
     parent: guides
 ---
 
-- [How to create a check that depends on an asset](#how-to-create-a-check-that-depends-on-an-asset)
-- [How to create a handler that depends on an asset](#how-to-create-a-handler-that-depends-on-an-asset)
-- [How to create a mutator that depends on an asset](#how-to-create-a-mutator-that-depends-on-an-asset)
+- [1. Download an asset definition from Bonsai](#1-download-an-asset-definition-from-bonsai)
```
Comment (rgeniesse) [inline content/sensu-go/5.2/guides/install-check-executables-with-assets.md:15]: Noticed that these items have numbers to them. Is that intentional? I just haven't see us use it anywhere else.

File: content/sensu-go/5.2/guides/install-check-executables-with-assets.md [hunk 2]
```diff
@@ -10,153 +12,119 @@ menu:
...
 
-## Why use assets?
-When configuration management is unavailable, assets can help manage runtime 
-dependencies such as scripts (ex: check-haproxy.sh) and tar files (ex: sensu-ruby-runtime.tar.gz)
-entirely within Sensu. 
+### 1. Download an asset definition from Bonsai
 
-## How to create a check that depends on an asset 
+You can discover, download, and share assets using [Bonsai, the Sensu asset index][16].
+To use an asset, select the Download button in Bonsai and download the asset definition for your platform and architecture.
```
Comment (rgeniesse) [inline content/sensu-go/5.2/guides/install-check-executables-with-assets.md]:
```suggestion
To use an asset, select the Download button on an asset's page in Bonsai and select the asset definition for your platform and architecture.
```

File: content/sensu-go/5.2/reference/assets.md [hunk 3]
```diff
@@ -41,18 +59,120 @@ If the requested asset is not in the local cache, it is downloaded from the asse
...
-The agent expects that an asset is a `TAR` archive that may optionally be
-GZip'd. Any scripts or executables should be within a `bin/` folder within in
-the archive.
+### Example structure
+{{< highlight shell >}}
+sensu-example-handler_1.0.0_linux_amd64
+├── CHANGELOG.md
+├── LICENSE
+├── README.md
+└── bin
```
Comment (rgeniesse) [inline content/sensu-go/5.2/reference/assets.md:68]: Maybe show some script inside `bin`, like `my-check.sh`, to further hammer home where the executables go.

Although, do we then do that with `lib` and `include`?

File: content/sensu-go/5.2/reference/assets.md [hunk 4]
```diff
@@ -41,18 +59,120 @@ If the requested asset is not in the local cache, it is downloaded from the asse
...
-- `{PATH_TO_ASSET}/bin` is injected into the `PATH` environment variable.
-- `{PATH_TO_ASSET}/lib` is injected into the `LD_LIBRARY_PATH` environment
-  variable.
-- `{PATH_TO_ASSET}/include` is injected into the `CPATH` environment variable.
+Share your open-source assets on [Bonsai][16] and connect with the Sensu Community.
+Bonsai supports assets hosted on [GitHub](https://github.com) and released using [GitHub releases](https://help.github.com/articles/about-releases/).
+For more information about creating Sensu Plugins, see the [Sensu Plugin specification][29].
+
+Bonsai requires a [`bonsai.yml` configuration file](#bonsai-yml-specification) in the root directory of your repository that includes the project description, platforms, asset filenames, and SHA-512 checksums.
+For a Bonsai-compatible asset template using Go and [GoReleaser](https://goreleaser.com/), see the [Sensu Go plugin skeleton][28].
```
Comment (rgeniesse) [inline content/sensu-go/5.2/reference/assets.md:80]: Maybe call out in some fashion how non-Go assets would work or link to that content if already created.

File: content/sensu-go/5.2/reference/assets.md [hunk 5]
```diff
@@ -41,18 +59,120 @@ If the requested asset is not in the local cache, it is downloaded from the asse
...
+asset_filename | 
+-------------|------
+description  | The filename of the archive containing the asset
+required     | true
+type         | String
+example      | {{< highlight yml >}}asset_filename: "#{repo}_#{version}_linux_amd64.tar.gz"{{< /highlight >}}
+
+sha_filename | 
+-------------|------
+description  | The SHA-512 checksum for the archive
```
Comment (rgeniesse) [inline content/sensu-go/5.2/reference/assets.md]:
```suggestion
description  | The SHA-512 checksum for the asset archive
```

Not sure if that is the proper language. I understand an asset to be the entire package, not just the executable(s) inside. Suggestion made as all other attribute descriptions under the `spec` scope contain it.

**Review by rgeniesse — CHANGES_REQUESTED** [review]
Looks great! Few suggestions, feel free to toss out if they don't fit.

I don't have any immediate suggestions for the asset creation section, but I'll continue thinking about it.

File: content/sensu-go/5.2/reference/assets.md (hunk 3 above)
Comment (apaskulin) [reply content/sensu-go/5.2/reference/assets.md:68]: That would be great. Would that be `my-check.sh` or just `my-check`?

File: content/sensu-go/5.2/reference/assets.md (hunk 4 above)
Comment (apaskulin) [reply content/sensu-go/5.2/reference/assets.md:80]: Good point. How do they work?

File: content/sensu-go/5.2/_index.md [hunk 6]
```diff
@@ -47,8 +47,8 @@ Sensu agents automatically register and de-register themselves with the Sensu se
 
 Get meaningful alerts when and where you need them.
 Use [event filters][18] to reduce noise and [check hooks][19] to add context and speed up incident response.
-Sensu integrates with the tools and services your organization already uses like [InfluxDB][6], [Slack][7], and more.
-Check out the [200+ plugins shared by the Sensu community][8], or write your own [Sensu Plugins][9] in any language.
+Sensu integrates with the tools and services your organization already uses like [PagerDuty][28], [AWS][29], and more.
```
Comment (cwjohnston) [inline content/sensu-go/5.2/_index.md]:
```suggestion
Sensu integrates with the tools and services your organization already uses like [PagerDuty][28], [InfluxDB][29], and more.
```

File: content/sensu-go/5.2/guides/install-check-executables-with-assets.md (hunk 1 above)
Comment (apaskulin) [reply content/sensu-go/5.2/guides/install-check-executables-with-assets.md:15]: Yep, it's intentional. In this case, it's a very 1-2-3 kind of process.

File: content/sensu-go/5.2/_index.md [hunk 7]
```diff
@@ -89,3 +89,6 @@ permissive [MIT License][12] and publicly available on [GitHub][13].
 [25]: dashboard/overview
 [26]: sensuctl/reference
 [27]: reference/handlers
+[28]: https://bonsai.sensu.io/assets/sensu/sensu-pagerduty-handler
+[29]: https://bonsai.sensu.io/assets/sensu/sensu-aws
```
Comment (cwjohnston) [inline content/sensu-go/5.2/_index.md:93]:
```suggestion
[29]: https://bonsai.sensu.io/assets/sensu/sensu-influxdb-handler
```

File: content/sensu-go/5.2/_index.md (hunk 6 above)
Comment (cwjohnston) [reply content/sensu-go/5.2/_index.md]: I know it's not on bonsai at this moment, but I'd like to keep the influxdb example. The documentation for that collection of AWS plugins just doesn't pass muster. I am working on getting a release cut and onto bonsai.

File: content/sensu-go/5.2/guides/install-check-executables-with-assets.md [hunk 8]
```diff
@@ -10,153 +12,119 @@ menu:
...
 
-## Why use assets?
-When configuration management is unavailable, assets can help manage runtime 
-dependencies such as scripts (ex: check-haproxy.sh) and tar files (ex: sensu-ruby-runtime.tar.gz)
-entirely within Sensu. 
+### 1. Download an asset definition from Bonsai
 
-## How to create a check that depends on an asset 
+You can discover, download, and share assets using [Bonsai, the Sensu asset index][16].
+To use an asset, select the Download button on an asset's page in Bonsai and select the asset definition for your platform and architecture.
```
Comment (apaskulin) [inline content/sensu-go/5.2/guides/install-check-executables-with-assets.md]:
```suggestion
To use an asset, select the Download button on the asset page in Bonsai to download the asset definition for your platform and architecture.
```

File: content/sensu-go/5.2/reference/assets.md (hunk 3 above)
Comment (cwjohnston) [reply content/sensu-go/5.2/reference/assets.md:68]: I think either file name works, but maybe use `.sh`  example as it is more illustrative that the file is meant to be executable

File: content/sensu-go/5.2/guides/install-check-executables-with-assets.md [hunk 9]
```diff
@@ -10,153 +12,119 @@ menu:
...
-You now know how to create an asset and add it to a check, handler, and mutator as a dependency.
-For further reading, check out the [assets reference][1].
+See the [asset reference](../../reference/assets) more information about creating and sharing assets.
 
 [1]: ../../reference/assets/
-[2]: #creating-an-asset
\ No newline at end of file
+[2]: #creating-an-asset
+[3]: https://bonsai.sensu.io
+[4]: https://bonsai.sensu.io/assets/sensu/sensu-aws
```
Comment (cwjohnston) [inline content/sensu-go/5.2/guides/install-check-executables-with-assets.md:123]:
```suggestion
```

Comment (cwjohnston) [reply content/sensu-go/5.2/guides/install-check-executables-with-assets.md:123]: unused link afaict

File: content/sensu-go/5.2/reference/assets.md (hunk 4 above)
Comment (cwjohnston) [reply content/sensu-go/5.2/reference/assets.md:80]: AFAICT, the bonsai web app receives webhook events from github that tell it to update the index entry for the corresponding asset, an action which I think is initiated by the creation of the release itself, not the files associated, so there's no direct dependency on using goreleaser.

my understanding is that goreleaser's function is to attach (upload) assets built by travis to [GitHub releases](https://help.github.com/articles/about-releases/).

anyone not using goreleaser can manually create a release on github and upload files or, use some other tool to automate these tasks.

File: content/sensu-go/5.2/_index.md (hunk 7 above)
Comment (apaskulin) [reply content/sensu-go/5.2/_index.md:93]: That's not a valid link, but we can update this to reference the Prometheus collector instead.

File: content/sensu-go/5.2/reference/assets.md (hunk 4 above)
Comment (apaskulin) [reply content/sensu-go/5.2/reference/assets.md:80]: So if we want to say something like "Assets can be written in any language as long as..." How does that sentence end?

**Pull request merged**